### ISSUE TITLE
Builds failing with error: use of undeclared identifier 'swprintf_s'

### ISSUE THREAD

**Issue by elinor-fung** [dotnet/runtime#77282]
Likely from https://github.com/dotnet/runtime/pull/77228

Example: https://dev.azure.com/dnceng-public/public/_build/results?buildId=58398&view=logs&jobId=c15592f9-a6ec-52c9-e864-c0455e061f15&j=c15592f9-a6ec-52c9-e864-c0455e061f15&t=e6ee899e-01c1-5f09-519b-c43bc88cb55e
```
[ 84%] Building CXX object dlls/dbgshim/CMakeFiles/dbgshim.dir/dbgshim.cpp.o
  /Users/runner/work/1/s/src/mono/dlls/dbgshim/dbgshim.cpp:64:5: error: use of undeclared identifier 'swprintf_s'
      swprintf_s(w_android_run_adb_command, 2048, W("%ws %ws"), w_android_adb_path, w_command_to_execute);
      ^
  1 error generated.
  make[2]: *** [dlls/dbgshim/CMakeFiles/dbgshim.dir/dbgshim.cpp.o] Error 1
  make[1]: *** [dlls/dbgshim/CMakeFiles/dbgshim.dir/all] Error 2
  make[1]: *** Waiting for unfinished jobs....
```

@AaronRobinsonMSFT

**Comment by AaronRobinsonMSFT** [comment]
FYI @janvorli and @jkotas We are again hitting the PAL being pulled into unexpected places.

**Comment by BruceForstall** [comment]
Maybe eng\pipelines\common\evaluate-default-paths.yml needs to be updated so the right tests get triggered for mono?

**Comment by am11** [comment]
> unexpected places.

Mono debug shim is there since .NET 6 preview 4: https://github.com/dotnet/runtime/commit/82e8d0e8656c7b628fea21b7ed6753b920a075c0

**Comment by elinor-fung** [comment]
> Maybe eng\pipelines\common\evaluate-default-paths.yml needs to be updated so the right tests get triggered for mono?

Yeah, I think we should update at least https://github.com/dotnet/runtime/blob/a46a28e47c72155d4daab535e6846ae48fd6784c/eng/pipelines/common/evaluate-default-paths.yml#L61 to trigger on src/coreclr/pal changes.

**Comment by jkotas** [comment]
It should be on all src/coreclr changes. The Mono dbg shim depends on like number of different pieces from src/coreclr

**Comment by elinor-fung** [comment]
Ah, then I guess it is more we need to stop excluding src/coreclr
https://github.com/dotnet/runtime/blob/a46a28e47c72155d4daab535e6846ae48fd6784c/eng/pipelines/common/evaluate-default-paths.yml#L79

**Comment by AaronRobinsonMSFT** [comment]
Fix is up - https://github.com/dotnet/runtime/pull/77284

Please someone tell me how I should update evaluate-default-paths.yml. I can add that to the PR.

**Comment by elinor-fung** [comment]
Or maybe it could be narrowed down to 
clrdefinitions.cmake
inc/
md/
minipal/
nativeresources/
pal/
palrt/
utilcode/

Based on a coarse search for CLR_DIR in mono.
